src/main/java/com/codeborne/selenide/conditions/SelectedText.java

```diff
@@ -0,0 +1,33 @@
+package com.codeborne.selenide.conditions;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.Driver;
+import org.openqa.selenium.WebElement;
+
+public class SelectedText extends Condition {
+  private final String expectedText;
+
+  public SelectedText(String expectedText) {
+    super("selectedText");
+    this.expectedText = expectedText;
+  }
+
+  private String actualResult = "";
+
+  @Override
+  public boolean apply(Driver driver, WebElement element) {
+    actualResult = driver.executeJavaScript(
+      "return arguments[0].value.substring(arguments[0].selectionStart, arguments[0].selectionEnd);", element);
+    return actualResult.equals(expectedText);
+  }
+
+  @Override
+  public String actualValue(Driver driver, WebElement element) {
+    return "'" + actualResult + "'";
+  }
+
+  @Override
+  public String toString() {
+    return getName() + " '" + expectedText + '\'';
```

Comment:
why sometimes `"'"` and sometimes `'\''`?

Reply:
@muescha  It's because IDEA sometimes suggest to replace one-characters String (like `"x"`) to a character (like `'x'`). I guess it's some kind of memory optimization. 

In this case, optimization is not needed, and `"'"` is more readable than `'\''`. So I replaces `'\''` everywhere in the code (see https://github.com/selenide/selenide/commit/1dbb92915844d6190e1d7f56e4e61602292852dc).